packer/linux/conf/bin/bk-fetch.sh

```diff
@@ -1,13 +1,66 @@
 #!/bin/bash
 set -euo pipefail
 
+# fetch_ssm_parameters fetches all SSM parameters under the given path and writes them to the given output file
+fetch_ssm_parameters() {
+  local ssm_path="$1"
+  local output_file="$2"
+
+  # check if the ssm_path is set
+  if [ -z "$ssm_path" ]; then
+    echo "ssm_path is not set"
+    return 1
+  fi
+
+  # check if the output_file is set
+  if [ -z "$output_file" ]; then
+    echo "output_file is not set"
+    return 1
+  fi
+
+  # trim off ssm: prefix
+  ssm_path=${ssm_path//ssm:/}
```

Comment:
you could make this a valid URI by using file-like syntax meaning that a valid SSM URI would have three slashes (`/`) like `ssm:///path/to/traverse`:

```suggestion
  # trim off ssm:// prefix
  ssm_path=${ssm_path//ssm:\/\//}
```